src/pytorch_lightning/utilities/argparse.py

```diff
@@ -38,8 +40,8 @@ def parse_argparser(cls, args: "ArgumentParser") -> Any:
 
 
 def from_argparse_args(
-    cls: Type[ParseArgparserDataType], args: Union[Namespace, ArgumentParser], **kwargs: Any
-) -> ParseArgparserDataType:
+    cls: FROM_ARGPARSE_ARGS_CLS, args: Union[Namespace, ArgumentParser], **kwargs: Any
```

Comment:
original error this change is meant to resolve:

```sh
src/pytorch_lightning/core/datamodule.py:98: error: Argument 1 to "from_argparse_args" has incompatible type "Type[LightningDataModule]"; expected "Type[ParseArgparserDataType]"  [arg-type]
```

Reply:
I have to say, I'm not sure about this